Composer/packages/lib/code-editor/src/utils/lspUtil.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as normalizeUrl from 'normalize-url';
+import ReconnectingWebSocket from 'reconnecting-websocket';
+
+export function createUrl(server: { [key: string]: string } | string): string {
+  if (typeof server === 'string') {
+    return normalizeUrl.default(server).replace(/^http/, 'ws');
+  }
+  const { host, hostname = location.hostname, port = location.port, path = '/' } = server;
+  const protocol = location.protocol === 'https:' ? 'wss' : 'ws';
+  const endHost = host || `${hostname}:${port}`;
+  return normalizeUrl.default(`${protocol}://${endHost}/${path}`);
+}
+
+export function createWebSocket(url: string): WebSocket {
+  const socketOptions = {
+    constructor: WebSocket,
+    maxReconnectionDelay: 10000,
+    minReconnectionDelay: 1000,
+    reconnectionDelayGrowFactor: 1.3,
+    connectionTimeout: 10000,
+    maxRetries: Infinity,
```

Comment:
let's not make this Infinity. How about 500?

Reply:
If not Infinity,  for some scenarios, e.g
1. computer re-awake from sleep, 
2. lost network then reconnected, 

editor will lost connection with lsp server and user have to refresh page.
this will be ok?